sycl/plugins/level_zero/pi_level_zero.hpp

```diff
@@ -94,6 +102,16 @@ struct _pi_platform {
   // Current number of L0 Command Lists created on this platform.
   // this number must not exceed ZeMaxCommandListCache.
   std::atomic<int> ZeGlobalCommandListCount{0};
+
+  // We need to store all memory allocations in the platform because there could
+  // be kernels with indirect access. Kernels with indirect access start to
+  // reference all existing memory allocations at the time when they are
+  // submitted to the device. Referenced memory allocations can be released only
+  // when kernel has finished execution.
+  // TODO: this container can be moved to _pi_context when Level Zero will
```

Comment:
Can we from the beginning have it be in pi_context since memory is allocated against a context?

Reply:
While needed we'd capture all contexts in the platform, such that you could deal with allocations at platform level.

Reply:
I can move it to pi_context, but mutex bust be kept in pi_platform, otherwise, kernel  will not be able to properly capture memory allocations from different contexts. Is it a good idea to separate mutex and container, or should I keep them together for now?

Reply:
I thought you would have a list of context in the platform (guarded with its own mutext). The list of allocation would reside in the context (also guarded with its own mutext).

Reply:
Implemented.